crates/libcontainer/src/container/builder.rs

```diff
@@ -50,7 +53,7 @@ impl<'a> ContainerBuilder<'a> {
     /// ```
     pub fn new(container_id: String, syscall: &'a dyn Syscall) -> Self {
         let root_path = PathBuf::from("/run/youki");
-
+        Self::validate_id(container_id.clone()).unwrap();
```

Comment:
I like the idea of validating the ID, but perhaps this check should happen before getting to the builder.

Reply:
Please take a look, it has been modified to perform verification before creating the builder.